File: src/minecraft/kevin/milk/utils/key/ClickUtils.java
package kevin.milk.utils.key;

public class ClickUtils {
    public static boolean isClickableWithRect(double x, double y, double dx, double dy, double mx, double my){
        return mx >= x && mx <= x + dx && my >= y && my <= y + dy;
    }
    public static boolean isClickable(double x, double y, double dx, double dy, double mx, double my){
        return mx >= x && mx <= dx && my >= y && my <= dy;
    }
    public static boolean isClickable(float x, float y, float dx, float dy, int mx, int my){
        return mx >= x && mx <= dx && my >= y && my <= dy;
    }
}

File: src/minecraft/kevin/utils/ClickUtils.kt
package kevin.utils

import kevin.utils.Mc.Companion.mc
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.inventory.ClickType
import net.minecraft.network.play.client.CPacketClickWindow

object ClickUtils {
    fun windowClick(
        windowID: Int,
        slotID: Int,
        bottom: Int,
        clickType: ClickType,
        player: EntityPlayer
    ) {
        val short1 = player.openContainer.getNextTransactionID(player.inventory)
        val itemstack = player.openContainer.slotClick(
            slotID,
            bottom,
            clickType,
            player
        )
        mc.connection!!.sendPacket(
            CPacketClickWindow(
                windowID,
                slotID,
                bottom,
                clickType,
                itemstack,
                short1
            )
        )
    }
}